client/main.js

```diff
@@ -390,18 +390,40 @@ $(document).ready(function() {
   }
 
   $('#nav-wiki-btn').on('click', function(event) {
-      if (!(event.ctrlKey || event.metaKey)) {
-          toggleWiki();
-      }
+    if (!(event.ctrlKey || event.metaKey)) {
+      toggleWiki();
+    }
   });
 
   $('.wiki-aside-action-collapse').on('click', collapseWiki);
 
+  function getWikiPath() {
+    var challengeType = window.common.challengeType;
+    var dashedName = window.common.dashedName;
+    var prefix = '';
+    // Get wiki prefix depending on type of challenge
+    // Types 0, 1 typically have solutions on the wiki prefixed with challenge-
+    // Type 5 typically has a solution on the wiki prefixed with algorithm-
+    if (challengeType === '0' || challengeType === '1') {
+      prefix = 'challenge-';
+      return prefix + dashedName;
+    }
+    if (challengeType === '5') {
+      prefix = 'algorithm-';
+      return prefix + dashedName;
+    }
+    return false;
+  }
+
   function showWiki() {
     if (!main.isWikiAsideLoad) {
       var lang = window.location.toString().match(/\/\w{2}\//);
       lang = (lang) ? lang[0] : '/en/';
       var wikiURL = '//freecodecamp.github.io/wiki' + lang;
+      var wikiPath = getWikiPath();
+      if (wikiPath) {
+        wikiURL = '//freecodecamp.github.io/wiki/en/' + wikiPath;
```

Comment:
Yeah I see this challenge in implementing this. The wiki is served from the gh-pages on GitHub.
So technically option 3 above is not possible (there is no wiki server per say).

That leaves us with option 1 and 2.

Option 2 is an overload because its an additional request and probably as you rightly said for every page.

While I do not like option 1 it could be done for now, else people are never going to discover that we have wiki articles (however few now) in the language they want to use FCC.

/cc @FreeCodeCamp/issue-moderators  Or should we drop the entire thing and let this redirect to english always as currently implemented.